activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -72,6 +72,23 @@ class SQLite3Adapter < AbstractAdapter
         boolean:      { name: "boolean" }
       }
 
+      ##
+      # :singleton-method:
+      # Indicates whether boolean values are stored in sqlite3 databases as 1
+      # and 0 or 't' and 'f'. Leaving `ActiveRecord::ConnectionAdapters::SQLite3Adapter.represent_boolean_as_integer`
+      # set to false is deprecated. SQLite databases have used 't' and 'f' to
+      # serialize boolean values and must have old data converted to 1 and 0
+      # (its native boolean serialization) before setting this flag to true.
+      # Conversion can be accomplished by setting up a rake task which runs
+      #
+      #   ExampleModel.where("boolean_column = 't'").update_all(boolean_column: 1)
```

Comment:
Note for anyone else that might have run into this and have tried to migrate a databare from f/t to 0/1:

If your migration specified a default for your boolean column this will continue to produce f/t. The columns default value also needs to be updated.